src/wifipaf/WiFiPAFEndPoint.cpp

```diff
@@ -797,81 +789,83 @@ SequenceNumber_t WiFiPAFEndPoint::AdjustRemoteReceiveWindow(SequenceNumber_t las
 }
 
-CHIP_ERROR WiFiPAFEndPoint::GetPktSn(Encoding::LittleEndian::Reader & reader, uint8_t * pHead, SequenceNumber_t & seqNum)
+CHIP_ERROR WiFiPAFEndPoint::GetPktSn(Encoding::LittleEndian::Reader reader, SequenceNumber_t & seqNum)
 {
     CHIP_ERROR err;
-    BitFlags<WiFiPAFTP::HeaderFlags> rx_flags;
-    size_t SnOffset = 0;
-    SequenceNumber_t * pSn;
-    err = reader.Read8(rx_flags.RawStorage()).StatusCode();
-    if (rx_flags.Has(WiFiPAFTP::HeaderFlags::kHankshake))
+    BitFlags<WiFiPAFTP::HeaderFlags> rxFlags;
+    size_t SkipOffset = 0;
+
+    err = reader.Read8(rxFlags.RawStorage()).StatusCode();
+    SuccessOrExit(err);
+    if (rxFlags.Has(WiFiPAFTP::HeaderFlags::kHandshake))
     {
         // Handkshake message => No ack/sn
         return CHIP_ERROR_INTERNAL;
     }
     // Always has header flag
-    SnOffset += kTransferProtocolHeaderFlagsSize;
-    if (rx_flags.Has(WiFiPAFTP::HeaderFlags::kManagementOpcode)) // Has Mgmt_Op
+    if (rxFlags.Has(WiFiPAFTP::HeaderFlags::kManagementOpcode))
     {
-        SnOffset += kTransferProtocolMgmtOpSize;
+        SkipOffset += kTransferProtocolMgmtOpSize;
     }
-    if (rx_flags.Has(WiFiPAFTP::HeaderFlags::kFragmentAck)) // Has ack
+    if (rxFlags.Has(WiFiPAFTP::HeaderFlags::kFragmentAck))
     {
-        SnOffset += kTransferProtocolAckSize;
+        SkipOffset += kTransferProtocolAckSize;
     }
-    pSn    = pHead + SnOffset;
-    seqNum = *pSn;
-
-    return CHIP_NO_ERROR;
+    err = static_cast<Encoding::LittleEndian::Reader &>(reader.Skip(SkipOffset)).Read8(&seqNum).StatusCode();
+    SuccessOrExit(err);
+exit:
+    return err;
 }
 
-CHIP_ERROR WiFiPAFEndPoint::DebugPktAckSn(const PktDirect_t PktDirect, Encoding::LittleEndian::Reader & reader, uint8_t * pHead)
+CHIP_ERROR WiFiPAFEndPoint::DebugPktAckSn(const PktDirect_t pktDirect, Encoding::LittleEndian::Reader reader)
 {
 #ifdef CHIP_WIFIPAF_END_POINT_DEBUG_LOGGING_ENABLED
-    BitFlags<WiFiPAFTP::HeaderFlags> rx_flags;
+    BitFlags<WiFiPAFTP::HeaderFlags> rxFlags;
     CHIP_ERROR err;
-    uint8_t * pAct = nullptr;
-    char AckBuff[4];
-    uint8_t * pSn;
-    size_t SnOffset = 0;
+    bool hasAck = false;
+    SequenceNumber_t ackNum, snNum;
+    size_t SkipOffset = 0;
 
-    err = reader.Read8(rx_flags.RawStorage()).StatusCode();
-    SuccessOrExit(err);
-    if (rx_flags.Has(WiFiPAFTP::HeaderFlags::kHankshake))
+    err = reader.Read8(rxFlags.RawStorage()).StatusCode();
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+    if (rxFlags.Has(WiFiPAFTP::HeaderFlags::kHandshake))
     {
         // Handkshake message => No ack/sn
         return CHIP_NO_ERROR;
     }
-    // Always has header flag
-    SnOffset += kTransferProtocolHeaderFlagsSize;
-    if (rx_flags.Has(WiFiPAFTP::HeaderFlags::kManagementOpcode)) // Has Mgmt_Op
+
+    if (rxFlags.Has(WiFiPAFTP::HeaderFlags::kManagementOpcode))
     {
-        SnOffset += kTransferProtocolMgmtOpSize;
+        SkipOffset += kTransferProtocolMgmtOpSize;
```

Comment:
Instead of setting and then resetting the offset.... can't we just `reader.Skip(kTransferProtocolMgmtOpSize)` here?

Reply:
Yes, it's cleaner. I will change it